packages/cli/src/update-notifier.ts

```diff
@@ -25,137 +25,128 @@ export class UpdateNotifier extends Context.Tag("@cli/UpdateNotifier")<
-                if (isNewer(cached.latest, currentVersion)) {
-                  yield* logUpdate(currentVersion, cached.latest, pkgName);
-                }
-                return;
-              }
+              if (!res.ok) return Option.none<string>();
+              const json = (await res.json()) as { version?: string };
+              return json.version
+                ? Option.some(json.version)
+                : Option.none<string>();
+            } finally {
+              clearTimeout(timer);
             }
+          },
+          catch: () => Option.none<string>(),
+        }).pipe(Effect.catchAll(() => Effect.succeed(Option.none()))),
+    );
+
+    const logUpdate = Effect.fn("UpdateNotifier.logUpdate")(
+      (current: string, latest: string, pkgName: string) =>
+        Console.log(
+          [
+            `Update available for ${pkgName}: ${current} → ${latest}`,
+            `Run: bun add -g ${pkgName}@latest`,
+          ].join("\n"),
+        ),
+    );
 
-            const latest = yield* fetchLatest(pkgName);
-            if (Option.isNone(latest)) return;
+    const check = Effect.fn("UpdateNotifier.check")(
+      function* (pkgName: string, currentVersion: string) {
+        if (config.isCi) return;
 
-            const latestVersion = latest.value;
-            yield* writeCache(file, {
-              latest: latestVersion,
-              nextCheck: now + config.checkInterval,
-            });
+        const now = Date.now();
+        const file = cachePath(pkgName);
 
-            if (isNewer(latestVersion, currentVersion)) {
-              yield* logUpdate(currentVersion, latestVersion, pkgName);
+        const cache = yield* readCache(file);
+        if (Option.isSome(cache)) {
+          const cached = cache.value;
+          if (cached.nextCheck > now) {
+            if (isNewer(cached.latest, currentVersion)) {
+              yield* logUpdate(currentVersion, cached.latest, pkgName);
             }
-          }).pipe(Effect.catchAll(() => Effect.void)),
-      );
-
-      return UpdateNotifier.of({ check });
-    }),
-  );
-
-  static readonly testLayer = Layer.succeed(
-    UpdateNotifier,
-    UpdateNotifier.of({
-      check: (_pkgName, _currentVersion) => Effect.void,
-    }),
-  );
+            return;
+          }
+        }
+
+        const latest = yield* fetchLatest(pkgName);
+        if (Option.isNone(latest)) return;
+
+        const latestVersion = latest.value;
+        yield* writeCache(file, {
+          latest: latestVersion,
+          nextCheck: now + config.checkInterval,
+        });
+
+        if (isNewer(latestVersion, currentVersion)) {
+          yield* logUpdate(currentVersion, latestVersion, pkgName);
+        }
+      },
+      Effect.catchAll(() => Effect.void),
+    );
+
+    return UpdateNotifier.of({ check });
+  }).pipe(Layer.effect(UpdateNotifier));
+
+  static readonly testLayer = Layer.succeed(UpdateNotifier, {
+    check: (_pkgName, _currentVersion) => Effect.void,
+  });
```

Comment:
Alos ditto. But I don't mind removing `UpdateNotifier.of` :) I think we'll still get nice autocomplete based on its position.